lib/extras/codec_apng.cc

```diff
@@ -380,23 +585,69 @@ Status DecodeImageAPNG(const Span<const uint8_t> bytes,
             delete[] chunk.p;
             break;
           }
+        } else if (id == kId_iCCP) {
+          if (processing_data(png_ptr, info_ptr, chunk.p, chunk.size)) {
+            JXL_WARNING("Corrupt iCCP chunk");
+            delete[] chunk.p;
```

Comment:
Not for this MR since this was already used before, but it could be a good idea to use RAII for these resources

Reply:
https://en.cppreference.com/w/cpp/language/raii#:~:text=Resource%20Acquisition%20Is%20Initialization%20or,in%20limited%20supply)%20to%20the

(i.e. have destructors free the memory)

Reply:
Ah, yes, that is indeed a nicer way of doing it. I didn't know that acronym, and I read it as camelcase RALL (Remove All?), which confused me. Sans serif fonts have that downside...

Reply:
IMHO Homo sapiens should have improved their fonts already to avoid these ambiguities. But that's for another discussion… :-)